types/size.pp

```diff
@@ -0,0 +1,4 @@
+# @summary Size definition
+# @author Tobias @towo Wolter <tobias.wolter+voxpupuli@b1-systems.de>
+# @see https://docs.bareos.org/Configuration/CustomizingTheConfiguration.html#datatypesize
+type Bareos::Size = Pattern[/(?i:^(\d+(\.\d+)?)\W*(k|kb|m|mb|g|gb)$)/]
```

Comment:
These regular expressions should have comments that explain what they are attempting to match.

Reply:
And for this you can borrow some tests in the bacula module :wink: :
https://github.com/xaque208/puppet-bacula/blob/master/spec/type_aliases/size_spec.rb